build_runner/lib/src/server/server.dart

```diff
@@ -300,10 +302,19 @@ window.\$dartLoader.forceLoadModule('packages/build_runner/src/server/build_upda
 class AssetHandler {
   final FinalizedReader _reader;
   final String _rootPackage;
+  late final AssetId _notFoundDefaultsTo;
+  late final bool _redirectNotFound;
 
   final _typeResolver = MimeTypeResolver();
 
-  AssetHandler(this._reader, this._rootPackage);
+  AssetHandler(this._reader, this._rootPackage, {String? notFoundDefaultsTo}) {
```

Comment:
Lets avoid `late` fields here, they aren't necessary.

I would:

- make `_redirectNotFound` a getter (just checks if `_notFoundDefaultsTo` is null).
- make `notFoundDefaultsTo` be typed as `AssetId` directly, force the caller to construct it.